src/forklift/core.py

```diff
@@ -233,70 +190,99 @@ def _hash(crate, hash_path):
         #: truncate destination table since we are hashing for the first time
         arcpy.TruncateTable_management(crate.destination)
 
-    shape_token = 'SHAPE@'
-    is_table = _is_table(crate)
+    shape_token = 'SHAPE@WKT'
+    src_id_field = 'src_id'
 
     log.info('checking for changes...')
     #: finding and filtering common fields between source and destination
     fields = set([fld.name for fld in arcpy.ListFields(crate.destination)]) & set([fld.name for fld in arcpy.ListFields(crate.source)])
-    fields = _filter_fields(fields)
-
-    sql_clause = None
+    fields = _filter_fields(fields, crate)
 
     #: keep track of OID token in order to remove from hashing
-    att_hash_sub_index = None
-    if 'OID@' in fields:
-        att_hash_sub_index = -1
-        source_describe = arcpy.Describe(crate.source)
-        sql_clause = (None, 'ORDER BY {}'.format(source_describe.OIDFieldName))
+    primary_key_index = -1
+    sql_clause = (None, 'ORDER BY {}'.format(crate.source_primary_key))
 
-    if not is_table:
+    if not crate.is_table():
         fields.append(shape_token)
-        att_hash_sub_index = -2
+        primary_key_index = -2
+
+    changes = Changes(list(fields))
+    changes.table = crate.source
+    changes.fields.append('OID@')
```

Comment:
It will be last sometimes or second to last when source has geometry. The source fields in the search cursor need to match the fields for the temp_table insert cursor, except for the last temp_table field which will be the source primary key. It could be changed or moved, but it would affect the temp_table insert cursor in _hash and all the cursors that add new rows.